modules/functions.js

```diff
@@ -0,0 +1,116 @@
+    <div class="task ${task.completed ? 'completed' : ''}">
+    <input type="checkbox" ${task.completed ? 'checked' : ''}id="${task.index}">
+    <label for = "${task.index}"> "${task.description}"</label>
+    <div class="remove"><button class="btnremove"><i class="fa">&#xf014;</i></button></div>
+    <div class="remove"><button class="btnEdit"><i class="fa-solid fa-pen-to-square"></i></button></div>
+
+    </div> <hr>
+    `).join('');
+  const removeTask = (index) => {
+    for (let i = index + 1; i < tasks.length; i += 1) {
+      tasks[i].index -= 1;
+    }
+    tasks.splice(index, 1);
+    taskPopulate(tasks);
+    setStorage(tasks);
+  };
+
+  const btnRemove = document.querySelectorAll('.btnremove');
+  if (btnRemove) {
+    btnRemove.forEach((btn, index) => {
+      btn.addEventListener('click', () => {
+        removeTask(index);
+      });
+    });
+  }
+  const editTask = (index) => {
+    const label = document.querySelector(`label[for='${index + 1}']`);
+    const currentText = label.innerText.trim();
+    const input = document.createElement('input');
+    input.type = 'text';
+    input.className = 'txtInput';
+    input.value = currentText;
+    label.replaceWith(input);
+    input.focus();
+    const saveChanges = () => {
+      const newText = input.value.trim();
+      if (newText !== '' && newText !== currentText) {
+        const newLabel = document.createElement('label');
+        newLabel.htmlFor = index;
+        newLabel.innerText = newText;
+        input.replaceWith(newLabel);
+        tasks[index].description = newText;
+        setStorage();
+      } else {
+        label.replaceWith(label);
+      }
+    };
+    input.addEventListener('blur', saveChanges);
+    input.addEventListener('keydown', (event) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        saveChanges();
+      } else if (event.key === 'Escape') {
+        event.preventDefault();
+        label.replaceWith(label);
+      }
+    });
+  };
+
+  const btnEdit = document.querySelectorAll('.btnEdit');
+  if (btnEdit) {
+    btnEdit.forEach((btn, index) => {
+      btn.addEventListener('click', () => {
+        editTask(index);
+      });
+    });
+  }
+  const checkboxes = document.querySelectorAll('input[type=checkbox]');
+  checkboxes.forEach((checkbox) => {
+    checkbox.addEventListener('change', (event) => {
+      const index = parseInt(event.target.id, 10) - 1;
+      tasks[index].completed = event.target.checked;
+      setStorage();
+    });
+  });
+};
+export const getStorage = (task) => {
+  task = JSON.parse(localStorage.getItem('taskData')) || [];
+  return task;
+};
+
+export const addList = () => {
+  const taskDescription = taskEntry.value;
+
+  const newTask = {
+
+    description: taskDescription,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  getStorage(tasks);
+  tasks.push(newTask);
+  setStorage(tasks);
+  taskPopulate(tasks);
+  taskEntry.value = '';
+};
```

Comment:
- [ ] You have completed all the requirements but there is a bug in your project. whenever I add the todos, they get stored in local storage but when I refresh the page, I can still see them in local storage but after refreshing the page, if I interact with the todos in any way, the list in the local storage gets updated and previous todos get removed.
kindly consider fixing this bug.

https://user-images.githubusercontent.com/96683843/236521513-d3483f55-dd39-496d-9268-ad31d37b3dbb.mov